### PR TITLE
Check in source code

### PR DESCRIPTION
**Pull request by maglims**
check in SpeechTranslator source code for new York make faire

### CODE REVIEW

**Comment by bfjelds** [comment]
remove project.lock.json file ... project.json should be included

File: SpeechTranslator/AdmAuthentication.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+        public AdmAuthentication(string clientid, string clientsecret)
+        {
+            this.clientId = clientid;
+            this.clientSecret = clientsecret;
+
+            this.request = string.Format("grant_type=client_credentials&client_id={0}&client_secret={1}&scope=http://api.microsofttranslator.com", WebUtility.UrlEncode(clientId), WebUtility.UrlEncode(clientSecret));
+
+
+            // this.request = "grant_type=client_credentials&client_id=magicbunny888&client_secret=FVw2RUvCvP%2f%2fCaBomIQliFxqWIVM8B0angjeLOZvvDE%3d&scope=http://api.microsofttranslator.com";
```
Comment (bfjelds) [inline SpeechTranslator/AdmAuthentication.cs]: remove all references to your personal account

File: SpeechTranslator/AdmAuthentication.cs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            string webResponse = "URI test";
+            byte[] bytes = Encoding.UTF8.GetBytes(requestDetails);
+            Task<Stream> requestTask = webRequest.GetRequestStreamAsync();
+            using (Stream requestStream = requestTask.Result)
+            {
+                requestStream.Write(bytes, 0, bytes.Length);
+            }
+
+            Task<WebResponse> responseTask = webRequest.GetResponseAsync();
+            /*
```
Comment (bfjelds) [inline SpeechTranslator/AdmAuthentication.cs]: remove all commented code

File: SpeechTranslator/AdmAuthentication.cs [hunk 3]
```diff
@@ -0,0 +1,121 @@
+﻿using System;
```
Comment (bfjelds) [inline SpeechTranslator/AdmAuthentication.cs]: add license banner

File: SpeechTranslator/AdmAuthentication.cs [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    class AdmAuthentication
+    {      // public static readonly string DatamarketAccessUri = "https://datamarket.accesscontrol.windows.net/v2/OAuth2-13";
+        private string clientId;
+        private string clientSecret;
+        private string request;
+        private Task<AdmAccessToken> token;
+        private static System.Threading.Timer accessTokenRenewer;
+
+        //Access token expires every 10 minutes. Renew it every 9 minutes only.
+        // private const int RefreshTokenDuration = 9;
```
Comment (bfjelds) [inline SpeechTranslator/AdmAuthentication.cs]: why is this commented?  if not needed, remove it

File: SpeechTranslator/App.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+{
+    /// <summary>
+    /// Provides application-specific behavior to supplement the default Application class.
+    /// </summary>
+    sealed partial class App : Application
+    {
+        /// <summary>
+        /// Allows tracking page views, exceptions and other telemetry through the Microsoft Application Insights service.
+        /// </summary>
+        public static Microsoft.ApplicationInsights.TelemetryClient TelemetryClient;
```
Comment (bfjelds) [inline SpeechTranslator/App.xaml.cs]: remove applicationinsights code

File: SpeechTranslator/AudioCapturePermissions.cs [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        /// this method tests the privacy setting for microphone access for this application.
+        /// Note that this only checks the Settings->Privacy->Microphone setting, it does not handle
+        /// the Cortana/Dictation privacy check, however (Under Settings->Privacy->Speech, Inking and Typing).
+        /// 
+        /// Developers should ideally perform a check like this every time their app gains focus, in order to 
+        /// check if the user has changed the setting while the app was suspended or not in focus.
+        /// </summary>
+        /// <returns>true if the microphone can be accessed without any permissions problems.</returns>
+        /// 
+        public async static Task<bool> RequestMicrophonePermission()
```
Comment (bfjelds) [inline SpeechTranslator/AudioCapturePermissions.cs]: how are we requesting microphone permission?  there is no consent UI on Iot Core.  should this method be renamed?

File: SpeechTranslator/ConstantParam.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace SpeechTranslator
+{
+    class ConstantParam
+    {
+        public static readonly string DatamarketAccessUri = "https://datamarket.accesscontrol.windows.net/v2/OAuth2-13";
+        public const int RefreshTokenDuration = 9;
+
+        //Token Property
+        public static string clientid = "magicbunny888";
```
Comment (bfjelds) [inline SpeechTranslator/ConstantParam.cs]: do not include your personal accounts in the samples

File: SpeechTranslator/ConstantParam.cs [hunk 8]
```diff
@@ -0,0 +1,43 @@
+﻿#define RPI1
```
Comment (bfjelds) [inline SpeechTranslator/ConstantParam.cs]: add license banner

File: SpeechTranslator/Package.appxmanifest [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+<Package xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10" xmlns:mp="http://schemas.microsoft.com/appx/2014/phone/manifest" xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10" IgnorableNamespaces="uap mp">
+  <Identity Name="SpeechTranslator-OnlyEnglishInput" Publisher="CN=Microsoft" Version="1.0.0.0" />
+  <mp:PhoneIdentity PhoneProductId="a202e5e5-a722-4ad3-8b57-8b5d71b7c5a6" PhonePublisherId="00000000-0000-0000-0000-000000000000" />
+  <Properties>
+    <DisplayName>SpeechTranslator-OnlyEnglishInput</DisplayName>
+    <PublisherDisplayName>Microsoft</PublisherDisplayName>
+    <Logo>Assets\StoreLogo.png</Logo>
+  </Properties>
+  <Dependencies>
+    <TargetDeviceFamily Name="Windows.Universal" MinVersion="10.0.10069.0" MaxVersionTested="10.0.10069.0" />
```
Comment (bfjelds) [inline SpeechTranslator/Package.appxmanifest]: this isn't even close to the current version?

File: SpeechTranslator/Translator.cs [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+        public string GetTranslatedString()
+        {
+            return this.translated;
+        }
+
+        public static string TranslateMethod(string authToken, string originalS, string from, string to)
+        {
+            string text = originalS; //"你能听见我";
+            //string from = "en";
```
Comment (bfjelds) [inline SpeechTranslator/Translator.cs]: why is all of this commented code in here?  remove it if it isn't needed.

File: SpeechTranslator/project.json [hunk 11]
```diff
@@ -0,0 +1,21 @@
+{
+  "dependencies": {
+    "Microsoft.ApplicationInsights": "1.0.0",
```
Comment (bfjelds) [inline SpeechTranslator/project.json]: remove applicationinsights code

File: SpeechTranslator/speechtranslator.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
+---
```
Comment (bfjelds) [inline SpeechTranslator/speechtranslator.md]: this belongs in ms-iot/content, not ms-iot/samples.

File: SpeechTranslator/ConstantParam.cs [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+        /// </summary>
+        /// 
+
+        // Stream Socket parameters
+        public static string SelfPort = "8082";
+        public static string ClientPort = "8083";
+        public static string ServerHostname = "speechtransrpi2";
+#else
+        /// <summary>
+        ///  Uncomment if you are deploying to device with the name "rpi2"
```
Comment (turkycat) [inline SpeechTranslator/ConstantParam.cs]: Should we clarify the step here? the comment says to 'uncomment' but is wrapped around #if blocks.

File: SpeechTranslator/MainPage.xaml.cs [hunk 14]
```diff
@@ -0,0 +1,513 @@
...
+                    catch (Exception exception)
+                    {
+                        // If this is an unknown status it means that the error is fatal and retry will likely fail.
+                        if (SocketError.GetStatus(exception.HResult) == SocketErrorStatus.Unknown)
+                        {
+                            await Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, () =>
+                            {
+                                connStatus.Text += "Reconnection failed, Double check Host is on and try again";
+                            });
+                            //throw;
```
Comment (turkycat) [inline SpeechTranslator/MainPage.xaml.cs]: remove this comment

File: SpeechTranslator/MainPage.xaml.cs [hunk 15]
```diff
@@ -0,0 +1,513 @@
...
+                    {
+                        // The underlying socket was closed before we were able to read the whole data.
+                        return;
+                    }
+
+                    // Display the string.
+                    string text = reader.ReadString(actualStringLength);
+
+                    string InLang ="en";
+                    string outLang = "zh-CHS";
```
Comment (turkycat) [inline SpeechTranslator/MainPage.xaml.cs:114]: consider making these member variables for the class? I noticed we have these strings as well as four Language objects which actually only point to two objects (en_lang and InputLang). Are they all necessary?

**Comment by turkycat** [comment]
As a general comment, I've noticed that your source files have inconsistent formatting. I realize this is very nitpicky, but I think formatting consistency shows attention to detail and high quality. For example, there are lines with variable declaration and initialization not separated by spaces:
`var obj=new Object()`
and others that are
`var obj = new Object()`

Some methods also begin on the very next line (no whitespace) after another method's closing brace, where as others do not (have one empty line). VS auto-format abilities can fix these (and other) little details for you. You can set your formatting preferences in Tools > Options > Text Editor and then auto format your documents by selecting relevant sections (or select all) and type CTRL + K, F (without releasing CTRL)

File: SpeechTranslator/MainPage.xaml.cs (hunk 15 above)
Comment (maglims) [inline SpeechTranslator/MainPage.xaml.cs:114]: yeah, the problem is speechRecognizer and translator don't use the same language string; like for Chinese, the speechRecognizer accepts zh-CN, but the translator service accepts zh-CHS; And for English, speechRecognizer accepts en-US, but translator service uses en.

File: SpeechTranslator/MainPage.xaml.cs [hunk 16]
```diff
@@ -0,0 +1,512 @@
...
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+
+using Windows.Media.SpeechRecognition;
+using Windows.Media.SpeechSynthesis;
+using Windows.UI.Core;
+
+// Added By Maggie
```
Comment (bfjelds) [inline SpeechTranslator/MainPage.xaml.cs]: probably don't need to personally identify this bit as added by Maggie?

File: SpeechTranslator/MainPage.xaml.cs [hunk 17]
```diff
@@ -0,0 +1,512 @@
...
+using Windows.Networking;
+using Windows.Networking.Sockets;
+using Windows.Storage.Streams;
+using System.Text;
+
+
+
+// The Blank Page item template is documented at http://go.microsoft.com/fwlink/?LinkId=402352&clcid=0x409
+using Windows.ApplicationModel.Core;
```
Comment (bfjelds) [inline SpeechTranslator/MainPage.xaml.cs:16]: put these with the other usings above

File: SpeechTranslator/readme.md [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+- 1 Router 
+- 1 SD card reader
+
+
+### Setup your hardware
+- Follow the [instruction](http://ms-iot.github.io/content/en-US/win10/SetupRPI.htm) to flush your sd cards
+- Insert the sd card and USB headsets into devices; Also Connect HDMI monitors and Ethernet cable to the devices; You can connect both Ethernet cable to the router, then the router will be connected to the Internet
+- Once all connected, power the devices on
+- You should see the device name and device ip on the screen
+- Telnet/powershell into the device,
```
Comment (bfjelds) [inline SpeechTranslator/readme.md]: ssh/telnet/powershell

File: SpeechTranslator/readme.md [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+
+	Note: you can choose the other device name as well, but in constantParam.cs file(you will see this file once you download the sample and open the solution as below steps show), you need to match them up.
+	You will see the details later.
+
+- Restart the device: shutdown /r /t 0
+
+### Setup your sample:
+
+
+1. Download the sample from [here](https://github.com/ms-iot/samples/speechtranslator) to your local PC
```
Comment (bfjelds) [inline SpeechTranslator/readme.md]: this link doesn't contain the ability to simply download ... try pointing to the ms-iot/samples zip download (see the other samples in ms-iot/content) and direct them to download the zip of all samples and open the speech project.

File: SpeechTranslator/readme.md [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+2. Open the solution file in visual studio
+3. Open the constantParam.cs file, you need to specify your own clientid and clientsecret
+
+	Note: this is an azure account which you need to access the MS translation service in azure;
+	Follow this [link](http://www.microsoft.com/en-us/translator/getstarted.aspx) to get a account,
```
Comment (bfjelds) [inline SpeechTranslator/readme.md]: "to get an account" not "to get a account"

File: SpeechTranslator/readme.md [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+### Setup your sample:
+
+
+1. Download the sample from [here](https://github.com/ms-iot/samples/speechtranslator) to your local PC
+2. Open the solution file in visual studio
+3. Open the constantParam.cs file, you need to specify your own clientid and clientsecret
+
+	Note: this is an azure account which you need to access the MS translation service in azure;
+	Follow this [link](http://www.microsoft.com/en-us/translator/getstarted.aspx) to get a account, 
+	And	once you get the account, replace the clientid and clientsecret with yours.
```
Comment (bfjelds) [inline SpeechTranslator/readme.md]: formatting here ... too many spaces: "And     once"

File: SpeechTranslator/readme.md [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+	If you are going to deploy to the device `speechtransrpi1`, open the constantParam.cs file, make sure the serviceHostName is equal to `speechtransrpi2`;
+	Then right click on the project, under property/debug, put the ip address of `speechtransrpi1`.
+	
+
+	If you are going to deploy to the device `speechtransrpi2`, open the constantParam.cs file, make sure the serviceHostName is equal to `speechtransrpi1`;
+	Then right click on the project, under property/debug, put the ip address of `speechtransrpi2`.
+	
+	Basically, the demo works like a phone line. You need to specify the ServerHostName which is where you are going to send data to.
+	
+5. Once deployment is done, open the webb url which should be something like this: http://yourdeivceipaddress:8080, under App/Installed APP, choose the `speechtranslator` app,
```
Comment (bfjelds) [inline SpeechTranslator/readme.md]: if you are using VS to deploy, why not use F5 and actually start the app?  then tell the users that they can use webb as well.

File: SpeechTranslator/readme.md [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+	If you are going to deploy to the device `speechtransrpi2`, open the constantParam.cs file, make sure the serviceHostName is equal to `speechtransrpi1`;
+	Then right click on the project, under property/debug, put the ip address of `speechtransrpi2`.
+	
+	Basically, the demo works like a phone line. You need to specify the ServerHostName which is where you are going to send data to.
+	
+5. Once deployment is done, open the webb url which should be something like this: http://yourdeivceipaddress:8080, under App/Installed APP, choose the `speechtranslator` app, 
+	then click start;
+	Do this for the other deivce as well.
+	
+	Now you are done with repro the demo. and Feel free to try out the demo to see how it works
```
Comment (bfjelds) [inline SpeechTranslator/readme.md]: this sentence needs work...maybe simplify: "Now you are ready to use the speech translator!"

File: SpeechTranslator/MainPage.xaml.cs [hunk 24]
```diff
@@ -0,0 +1,512 @@
...
+                    // Display the string.
+                    string text = reader.ReadString(actualStringLength);
+
+                    string InLang ="en";
+                    string outLang = "zh-CHS";
+                    Translator Trans = new Translator(text, InLang, outLang);
+                    string translatedS = Trans.GetTranslatedString();
+
+                    SpeechSynthesisStream stream = await synthesizer.SynthesizeTextToStreamAsync(translatedS);
+                    var ignored = Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, () =>
```
Comment (bfjelds) [inline SpeechTranslator/MainPage.xaml.cs]: Windows.UI.Core is in the usings, no need to specify

File: SpeechTranslator/MainPage.xaml.cs [hunk 25]
```diff
@@ -0,0 +1,512 @@
...
+                }
+            }
+            catch (Exception exception)
+            {
+                // If this is an unknown status it means that the error is fatal and retry will likely fail.
+                if (SocketError.GetStatus(exception.HResult) == SocketErrorStatus.Unknown)
+                {
+                    throw;
+                }
+                var ignored = Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, () =>
```
Comment (bfjelds) [inline SpeechTranslator/MainPage.xaml.cs]: Windows.UI.Core is in the usings, no need to specify

File: SpeechTranslator/MainPage.xaml.cs [hunk 26]
```diff
@@ -0,0 +1,512 @@
...
+                    synthesizer.Voice = voice;
+                    break;
+                }
+            }
+
+            // Check Microphone Plugged in
+            bool permissionGained = await AudioCapturePermissions.RequestMicrophonePermission();
+            if (!permissionGained)
+            {
+                this.dictationTextBox.Text = "Permission to access capture resources was not given by the user, reset the application setting in Settings->Privacy->Microphone.";
```
Comment (bfjelds) [inline SpeechTranslator/MainPage.xaml.cs]: this comment is not applicable to IoT Core ... there is no settings.  there is no idea of consent on IoT Core.

File: SpeechTranslator/MainPage.xaml.cs [hunk 27]
```diff
@@ -0,0 +1,512 @@
...
+
+                    StreamSocket locsocket = new StreamSocket();
+
+                    // save the soket so subsequence can use it
+                    CoreApplication.Properties.Add("clientSocket", locsocket);
+                    try
+                    {
+                        await locsocket.ConnectAsync(hostName, ConstantParam.ClientPort);
+
+                        await Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, () =>
```
Comment (bfjelds) [inline SpeechTranslator/MainPage.xaml.cs]: Windows.UI.Core is in the usings, no need to specify

File: SpeechTranslator/MainPage.xaml.cs [hunk 28]
```diff
@@ -0,0 +1,512 @@
...
+                            connStatus.Text = "Connected Successfully!";
+                        });
+                        connected = true;
+                    }
+                    catch (Exception exception)
+                    {
+                        // If this is an unknown status it means that the error is fatal and retry will likely fail.
+                        if (SocketError.GetStatus(exception.HResult) == SocketErrorStatus.Unknown)
+                        {
+                            await Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, () =>
```
Comment (bfjelds) [inline SpeechTranslator/MainPage.xaml.cs]: Windows.UI.Core is in the usings, no need to specify

File: SpeechTranslator/MainPage.xaml.cs [hunk 29]
```diff
@@ -0,0 +1,512 @@
...
+                {
+                    dictationTextBox.Text = dictatedTextBuilder.ToString();
+                });
+            }
+
+        }
+
+        private async void SpeechRecognizer_StateChanged(SpeechRecognizer sender, SpeechRecognizerStateChangedEventArgs args)
+        {
+            await Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
```
Comment (bfjelds) [inline SpeechTranslator/MainPage.xaml.cs]: what is this doing?  can it be removed?

File: SpeechTranslator/MainPage.xaml.cs [hunk 30]
```diff
@@ -0,0 +1,512 @@
...
+        {
+            await Windows.System.Launcher.LaunchUriAsync(new Uri("ms-settings:privacy-speechtyping"));
+        }
+
+        private async void btnReConnect_Click(object sender, RoutedEventArgs e)
+        {
+            if (connected)
+            {
+
+                await Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, () =>
```
Comment (bfjelds) [inline SpeechTranslator/MainPage.xaml.cs]: Windows.UI.Core is in the usings, no need to specify

File: SpeechTranslator/AudioCapturePermissions.cs (hunk 6 above)
Comment (bfjelds) [inline SpeechTranslator/AudioCapturePermissions.cs]: change this method name to RequestMicrophoneCapture ... there is no permission on Iot Core

File: SpeechTranslator/ConstantParam.cs [hunk 31]
```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+#define RPI1
+
+using System;
```
Comment (bfjelds) [inline SpeechTranslator/ConstantParam.cs]: clean up the usings of your cs files ... VS can remove unused usings easily.

File: SpeechTranslator/readme.md [hunk 32]
```diff
@@ -0,0 +1,59 @@
...
+
+1. Download the sample from [here](https://github.com/ms-iot/samples/speechtranslator) to your local PC
+2. Open the solution file in visual studio
+3. Open the constantParam.cs file, you need to specify your own clientid and clientsecret
+
+	Note: this is an azure account which you need to access the MS translation service in azure;
+	Follow this [link](http://www.microsoft.com/en-us/translator/getstarted.aspx) to get a account, 
+	And	once you get the account, replace the clientid and clientsecret with yours.
+
+4. Build the solutions and deploy to devices;
```
Comment (bfjelds) [inline SpeechTranslator/readme.md]: there is no mention of the #define RPI1 or RPI2 here???

File: SpeechTranslator/readme.md (hunk 22 above)
Comment (maglims) [inline SpeechTranslator/readme.md]: That is also ok. But F5 is slow for this sample. If deploy to device first, then start the deployed App on webb. It is much faster.

**Pull request merged**